### PR TITLE
Implement CircleCI

### PR DESCRIPTION
**Pull request by colinbruce**
#### What
Implement CircleCI
- Add an after hook to cucumber to take a screenshot if it fails

#### Ticket
[CCCD : Spike investigate kubernetes](https://dsdmoj.atlassian.net/browse/CBO-520)

#### Why
The cloud platform team are implementing kubernetes as a hosting solution and most teams are using CircleCI as their CI/CD tool.  Doing so for CCCD has the following advantages:
- Parallel running of tests, Circle can be configured to run different tests simultaneously, whereas Travis runs them in sequence, this will reduce our CI runs from ~30 minutes to ~15.
- Circle will also replace Jenkins when the full switch to kubernetes is undertaken
- Circle can store screenshots of failing cucumber tests for review after the run has finished

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+          PGUSER: postgres
+          RAILS_ENV: test
+          TZ: Europe/London
+      - image: circleci/postgres:9.6-alpine
+        environment:
+          POSTGRES_USER: postgres
+          POSTGRES_DB: cccd_test
+          POSTGRES_PASSWORD: ""
+
+  install_phantom_js: &install_phantom_js
```
Comment (jsugarman) [inline .circleci/config.yml]: do we need this now? have removed this from feature tests in favour of headless chrome since you started working on this

Comment (jsugarman) [reply .circleci/config.yml]: **Not a blocker** As discussed you, apparently, can append `-browsers-legacy` to the docker image tag name to include PhantomJS

https://circleci.com/docs/2.0/browser-testing/

**Not a blocker**
An to switch jasmine over to using headless chrome i found this started for 10
https://github.com/jasmine/jasmine-gem/issues/293

File: features/support/hooks.rb [hunk 2]
```diff
@@ -51,6 +51,11 @@
   Cucumber::Rails::Database.javascript_strategy = :truncation, { except: NON_TRUNCATED_TABLES }
 end
 
+After do |scenario|
+  name = scenario.location.file.gsub('features/','').gsub(/\.|\//, '-')
+  screenshot_image(name) if scenario.failed?
```
Comment (jsugarman) [inline features/support/hooks.rb:56]: 👍 cool

**Review by jsugarman — APPROVED** [review]
👍 LGTM - singular point of concern related to not testing the merged PR (travis' default behaviour) but can iterate on this as discussed.

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,235 @@
+version: 2
+references:
+  build_containers_config: &build_container_config
+    working_directory: ~/repo
+    docker:
+      - image: circleci/ruby:2.5.3-node-browsers
```
Comment (jsugarman) [inline .circleci/config.yml]: possibly could use `circleci/ruby:2.5.3-node-browsers-legacy` to include phantomJS
https://circleci.com/docs/2.0/browser-testing/

File: .circleci/config.yml (hunk 1 above)
Comment (colinbruce) [reply .circleci/config.yml]: the `node-browsers-legacy` fix worked 💥

**Pull request merged**